lib/paperclip/storage/ftp.rb

```diff
@@ -35,16 +35,18 @@ def to_file(style_name = default_style)
       end
 
       def flush_writes
-        with_ftp_servers do |servers|
-          servers.map do |server|
-            Thread.new do
-              @queued_for_write.each do |style_name, file|
-                remote_path = path(style_name)
-                log("saving ftp://#{server.user}@#{server.host}:#{remote_path}")
-                server.put_file(file.path, remote_path)
+        unless @queued_for_write.empty?
+          with_ftp_servers do |servers|
+            servers.map do |server|
+              Thread.new do
+                @queued_for_write.each do |style_name, file|
+                  remote_path = path(style_name)
+                  log("saving ftp://#{server.user}@#{server.host}:#{remote_path}")
+                  server.put_file(file.path, remote_path)
+                end
               end
-            end
-          end.each(&:join)
+            end.each(&:join)
+          end
```

Comment:
Thanks for the PR! 

Instead of wrapping the code in a huge `unless` block, I would prefer to just use an early return in the first line of the method:

```ruby
return if @queued_for_write.empty?
```